contentcuration/contentcuration/decorators.py

```diff
@@ -41,13 +41,13 @@ def wrap(request, *args, **kwargs):
         try:
             channel = Channel.objects.get(pk=kwargs['channel_id'])
 
-            if not channel.public and \
-                not channel.editors.filter(id=request.user.id).exists() and \
-                not channel.viewers.filter(id=request.user.id).exists() and \
-                not request.user.is_admin:
-                return render_to_response('unauthorized.html', context_instance=RequestContext(request))
+            if channel.public or \
+                channel.editors.filter(id=request.user.id).exists() or \
+                channel.viewers.filter(id=request.user.id).exists() or \
+                request.user.is_admin:
+                return function(request, *args, **kwargs)
 
-            return function(request, *args, **kwargs)
+            return render_to_response('unauthorized.html', context_instance=RequestContext(request))
         except ObjectDoesNotExist:
```

Comment:
My only request is if you can find a way to restrict this try-catch to the parts that would actually raise the error, i.e. the filter clauses. Although would they even raise an error since we use a filter?

Reply:
Yeah, filter won't throw, that's why we can (and have to) use .exists() on it. :)

Reply:
Since it won't throw it, let's just remove that try except clause then!

Reply:
filter won't throw, but the Channel.objects.get call at the top will throw if it doesn't find the channel. (e.g. /channels/idontexist/edit) That's why we return a not found in that case. (It's not returning 404 status right now but I was planning on doing all the 404 status changes and tests in one.)

Reply:
I see. Then let's restrict the try except to only that `Channel.objects.get` then if that's the only place it'll throw that error.